ludwig/datasets/__init__.py

```diff
@@ -81,16 +83,19 @@ def load_dataset_uris(
 
     Returns the input unmodified for any non-Ludwig datasets.
     """
-    dataset_out = dataset
-    training_set_out = training_set
-    validation_set_out = validation_set
-    test_set_out = test_set
+
+    dataset_out, training_set_out, validation_set_out, test_set_out = dataset, training_set, validation_set, test_set
+    # Check that any of the datasets begin with the `hf://` prefix denoting a Hugging Face dataset URI
+    # Hugging Face datasets should follow the naming convention `hf://<hf_id>--<hf_subsample>`
+    if _is_hf(dataset, training_set):
+        return _load_hf_datasets(dataset, training_set, validation_set, test_set, backend)
 
     # Check that any of the datasets begin with the `ludwig://` prefix denoting a Ludwig dataset URI
     if dataset is not None:
         if isinstance(dataset, str) and dataset.startswith(URI_PREFIX):
             dataset_out = _load_cacheable_dataset(dataset, backend)
-    elif training_set is not None:
+        return dataset_out, training_set_out, validation_set_out, test_set_out
```

Comment:
the logic before allowed for having a ringle return in the function, which is auspicable for readability